Homework/js/lesson-5-task-1-4.js

```diff
@@ -0,0 +1,263 @@
+    }
+
+    const newLetterArray = [];
+    let i = 0;
+    while (i < letterArray.length) {
+        const ONE_ELEMENT_OF_ARRAY = letterArray[i].toUpperCase();
+        if (letterArray[i] === ONE_ELEMENT_OF_ARRAY) {
+            newLetterArray.push(letterArray[i].toLowerCase());
+        } else {
+            newLetterArray.push(letterArray[i].toUpperCase());
+        }
+        i++
+    }
+
+    return newLetterArray;
+}
+console.log(allLetterWhile(['a', 'B', 'C', 'L', 'g', 'g', 'EEE']));
+
+
+// for of
+const allLetterForOf = (letterArray) => {
+    if (!Array.isArray(letterArray)) {
+        return;
+    }
+
+    const newLetterArray = [];
+    for (const letter of letterArray) {
+        const ONE_ELEMENT_OF_ARRAY = letter.toUpperCase();
+        if (letter === ONE_ELEMENT_OF_ARRAY) {
+            newLetterArray.push(letter.toLowerCase());
+        } else {
+            newLetterArray.push(letter.toUpperCase());
+        }
+    }
+
+    return newLetterArray;
+}
+console.log(allLetterForOf(['a', 'B', 'C', 'L', 'GGGGg', 'g', 'EEE']));
+
+
+// map
+const allLetterMap = (letterArray) => {
+    if (!Array.isArray(letterArray)) {
+        return;
+    }
+
+    const stringNumbersArray = letterArray.map((val) => {
+        const ONE_ELEMENT_OF_ARRAY = val.toUpperCase();
+        if (val === ONE_ELEMENT_OF_ARRAY) {
+            return (val.toLowerCase());
+        } else {
+            return (val.toUpperCase());
+        }
+    })
+
+    return stringNumbersArray;
+}
+console.log(allLetterMap(['a', 'B', 'C', 'LJHHHL', 'GGGGg', 'g', 'EEE']));
+
+
+
+
+
+
+
+// 3)
+const numbersArrayFunction = (value) => {
+    if (!Array.isArray(value)) {
+        return;
+    }
+
+    const stringNumbersArray = []
+    for (let i = 0; i < value.length; i++) {
+        stringNumbersArray.push(`Value: ${value[i]}, Index: ${value.indexOf(value[i])}, Squared: ${Math.pow(value[i], 2)}`);
+    }
+
+    return stringNumbersArray;
+}
+console.log(numbersArrayFunction([3,4,6]));
+
+// while
+const numbersArrayFunctionWhile = (value) => {
+    if (!Array.isArray(value)) {
+        return;
+    }
+
+    let i = 0;
+    const stringNumbersArray = [];
+    while (i < value.length) {
+        stringNumbersArray.push(`Value: ${value[i]}, Index: ${value.indexOf(value[i])}, Squared: ${Math.pow(value[i], 2)}`);
+        i++;
+    }
+
+    return stringNumbersArray;
+}
+console.log(numbersArrayFunctionWhile([3,4,6]));
+
+
+// for of
+const numbersArrayFunctionForOf = (value) => {
+    if (!Array.isArray(value)) {
+        return;
+    }
+
+    const stringNumbersArray = [];
+    for (const val of value) {
+        stringNumbersArray.push(`Value: ${val}, Index: ${value.indexOf(val)}, Squared: ${Math.pow(val, 2)}`);
+    }
+
+    return stringNumbersArray;
+}
+console.log(numbersArrayFunctionForOf([3,4,6]));
+
+
+// map
+const numbersArrayFunctionMap = (value) => {
+    if (!Array.isArray(value)) {
+        return;
+    }
+
+    const stringNumbersArray = value.map((val) => {
```

Comment:
same as above about params names in map callback

Reply:
Corrected